Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_datetimeperiod.py

```diff
@@ -312,88 +312,104 @@ def is_valid_connector_for_date_and_time_period(self, text: str):
 
         return False
 
-    def match_time_of_day(self, source: str, reference: datetime, date_ers: [ExtractResult] = None):
-        tokens: List[Token] = list()
-        source = source.strip().lower()
+    def match_time_of_day(self, source: str, reference: datetime, date_extract_results: [ExtractResult] = None):
+        tokens = []
+        result = []
 
-        matches = regex.finditer(
-            self.config.specific_time_of_day_regex, source)
-        tokens.extend(map(lambda x: Token(x.start(), x.end()), matches))
+        matches = list(regex.finditer(self.config.specific_time_of_day_regex, source))
+        for match in matches:
+            tokens.append(Token(match.start(), match.end()))
 
-        ers_date: List[ExtractResult] = self.config.single_date_extractor.extract(
-            source, reference)
+        # Date followed by morning, afternoon or morning, afternoon followed by Date
+        if len(date_extract_results) == 0:
+            return tokens
 
-        for extracted_result in ers_date:
+        for extracted_result in date_extract_results:
             after_str = source[extracted_result.start + extracted_result.length:]
-            match = regex.search(
-                self.config.period_time_of_day_with_date_regex, after_str)
+
+            match = regex.search(self.config.period_time_of_day_with_date_regex, after_str)
 
             if match:
-                if not after_str[0:match.start()].strip():
-                    tokens.append(
-                        Token(extracted_result.start, extracted_result.start + extracted_result.length + match.end()))
-                else:
-                    pause_match = regex.search(
-                        self.config.middle_pause_regex, after_str[0:match.start()].strip())
-                    if pause_match:
-                        suffix = after_str[match.end():].strip()
+                # For cases like "Friday afternoon between 1PM and 4PM" which "Friday afternoon" need to be
+                # extracted first
+                match_start = match.start()
+                if not after_str[0:match.start()] or after_str[0:match.start()].isspace():
+                    start = extracted_result.start
+                    group = RegExpUtility.get_group_list(match, Constants.TIME_OF_DAY_GROUP_NAME)
+                    # TODO: + 1 has to be the index of the matching regex in the group
```

Comment:
Is ok this `TODO` sentence?

Reply:
It is. I will fix that before the PR is merged.